### PR TITLE
BUG: RollingGroupby when groupby key is in the index

### LINKED ISSUE #37641: BUG: RollingGroupby duplicates columns in index even with group_keys=False
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd

arrays = [['val1', 'val1', 'val2'], ['val1', 'val1', 'val2']]
index = pd.MultiIndex.from_arrays(arrays, names=('idx1', 'idx2'))

s = pd.Series([1, 2, 3], index=index)
print(
    s.groupby(["idx1", "idx2"], group_keys=False).rolling(1).mean()
)

```

#### Problem description

the group_keys parameter in Seried.groupby() is broken since 1.1.0 when using .rolling()

It does not seem to be an issue if another function instead of *rolling* is used, eg 
```python
    s.groupby(["idx1", "idx2"], group_keys=False).mean()
```
In pandas=1.1.{0,1} the output is

    idx1 idx2  
    val1 val1  (val1, val1)   1.0
         val1  (val1, val1)   2.0
    val2 val2  (val2, val1)   3.0

In pandas=1.1.{2,3,4} the output is

    idx1 idx2  idx1 idx2     
    val1 val1  val1 val1   1.0
                    val1   2.0
    val2 val2  val2 val2   3.0

#### Expected Output (as per pandas 1.0.*)

    idx1 idx2       
    val1 val1   1.0
         val1   2.0
    val2 val2   3.0

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f2ca0a2665b2d169c97de87b8e778dbed86aea07
python           : 3.6.7.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.17134
machine          : AMD64
processor        : Intel64 Family 6 Model 63 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.1
numpy            : 1.16.1
pytz             : 2018.9
dateutil         : 2.8.0
pip              : 10.0.1
setuptools       : 39.0.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.3.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10
IPython          : 7.2.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 2.6.0
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #37641
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by rhshadrach** [comment]
This will remove an index label in the case of partial overlap between the index labels and the groupby labels; I am wondering if a more natural behavior here would be "add any groupby labels to the index if they aren't already there".

**Comment by mroeschke** [comment]
Not sure if I 100% follow, but this is the resulting index behavior I am trying to mimic here:

```
In [1]: from pandas import *

In [2]:         arrays = [["val1", "val1", "val2"], ["val1", "val1", "val2"]]
   ...:         index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
   ...:
   ...:         df = DataFrame({"A": [1, 1, 2], "B": range(3)}, index=index)

In [3]: df
Out[3]:
           A  B
idx1 idx2
val1 val1  1  0
     val1  1  1
val2 val2  2  2

# idx2 is dropped
In [4]: df.groupby(["idx1", "A"]).mean()
Out[4]:
          B
idx1 A
val1 1  0.5
val2 2  2.0
```

**Comment by rhshadrach** [comment]
Shouldn't `groupby(...).rolling` behave more like a transform than a reduction? In any case, the behavior here does not mimic a reduction in the case when the group labels are entirely not in the index (please correct me if this is wrong!). This means we are mimicking the behavior of a reduction in one case but not another.

I almost never use rolling in my work so I don't have much of an opinion as to what the right behavior is. Just noticed what I _think_ is an inconsistency.

**Comment by mroeschke** [comment]
Yes groupby.rolling should essentially act as a transform, and I think in both test cases I added the resulting shape (number of rows) is maintained with consideration that not all columns should be maintained since one of the columns was a grouping key.

```
In [7]:         arrays = [["val1", "val1", "val2"], ["val1", "val1", "val2"]]
   ...:         index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
   ...:
   ...:         df = DataFrame({"A": [1, 1, 2], "B": range(3)}, index=index)

In [8]: result = df.groupby(["idx1", "A"]).rolling(1).mean()

In [9]: result
Out[9]:
          B
idx1 A
val1 1  0.0
     1  1.0
val2 2  2.0

In [10]: df
Out[10]:
           A  B
idx1 idx2
val1 val1  1  0
     val1  1  1
val2 val2  2  2
```

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -535,6 +535,7 @@ Groupby/resample/rolling
 - Bug in :meth:`df.groupby(..).quantile() <pandas.core.groupby.DataFrameGroupBy.quantile>` and :meth:`df.resample(..).quantile() <pandas.core.resample.Resampler.quantile>` raised ``TypeError`` when values were of type ``Timedelta`` (:issue:`29485`)
 - Bug in :meth:`Rolling.median` and :meth:`Rolling.quantile` returned wrong values for :class:`BaseIndexer` subclasses with non-monotonic starting or ending points for windows (:issue:`37153`)
 - Bug in :meth:`DataFrame.groupby` dropped ``nan`` groups from result with ``dropna=False`` when grouping over a single column (:issue:`35646`, :issue:`35542`)
+- Bug in :meth:`RollingGroupby` with the resulting :class:`MultiIndex` when grouping by a label that is in the index (:issue:`37641`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:538]: i think this is ok to backport ton 1.1.5 if possible

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.2.0.rst:538]: this fixes a regression from 1.0.5 -> 1.1.0, so backport is preferable.

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**